src/tools/tidy/src/ui_tests.rs

```diff
@@ -128,4 +132,44 @@ pub fn check(path: &Path, bad: &mut bool) {
             }
         }
     });
+
+    // Check that any test files modified don't have a useless name
+    for changed_file in git::get_git_modified_files(None, &vec!["rs"]).unwrap().unwrap() {
+        // `get_git_modified_files` reports all files in the repo, only check ui tests
+        if !paths.iter().any(|path| changed_file.starts_with(&path.to_string_lossy().into_owned()))
+        {
+            continue;
+        }
+
+        let changed_file = changed_file.as_str();
+
+        // use git rev-list to make sure that the file still exists, since `get_git_modified_files`
+        // names both the source and destination files in a rename.
+        let mut cmd = Command::new("git");
+        cmd.args(["rev-list", "-1", "--all"]);
+        cmd.arg(changed_file);
+
+        // If a non-zero status was returned, it's from a deleted file, otherwise it exists
+        let is_modified = cmd.output().expect("unable to execute `git rev-list`").status.success();
+
+        if is_modified {
+            lazy_static! {
+                static ref ISSUE_NAME_REGEX: Regex = Regex::new(r"issues?[-_]?\d+").unwrap();
+                static ref ICE_NAME_REGEX: Regex = Regex::new(r"ice[-_]?\d+").unwrap();
```

Comment:
I am not sure if we have a consensus against `ice-[0-9]+.rs` filenames, as we mentioned "random ICE stuff" as a case where it may be hard to find a good name, and I'd somewhat rather see e.g. `issue-[0-9]+-ice.rs`. Also, I only see a few tests that match this regex, and most would be a false positive:
```
tests/rustdoc-ui/intra-doc/span-ice-55723.rs
tests/ui/enum-discriminant/issue-70453-generics-in-discr-ice-2.rs
tests/ui/const-generics/generic_const_exprs/issue-97047-ice-2.rs
tests/ui/const-generics/generic_const_exprs/issue-97047-ice-1.rs
tests/ui/const-generics/inhabited-assoc-ty-ice-2.rs
tests/ui/const-generics/inhabited-assoc-ty-ice-1.rs
tests/ui/const-generics/ice-68875.rs
tests/ui/tuple/wrong_argument_ice-2.rs
tests/ui/tuple/wrong_argument_ice-3.rs
tests/ui/tuple/wrong_argument_ice-4.rs
tests/ui/consts/ice-48279.rs
tests/ui/nll/user-annotations/region-error-ice-109072.rs
tests/ui/array-slice-vec/slice-2.rs
tests/ui/moves/move-out-of-slice-1.rs
tests/ui/moves/move-out-of-slice-2.rs
```

So I think we can skip this one for now.